drivers/ipm/ipm_rpi_pico.c

```diff
@@ -0,0 +1,286 @@
+
+	/* Stack pointer shall point within RAM assigned to the core. */
+	if (!address_in_range(cpu1_sp, sram_start, sram_size)) {
+		LOG_ERR("CPU1 stack pointer 0x%08x invalid.", cpu1_sp);
+		return -EINVAL;
+	} else {
+		LOG_DBG("CPU1 stack pointer: 0x%08x", cpu1_sp);
+	}
+#endif /* DT_INST_NODE_HAS_PROP(0, memory_region) */
+
+	/* Initial program counter shall point to code within the CPU1 partition. */
+	if (!address_in_range(cpu1_pc, DT_REG_ADDR(DT_INST_PHANDLE(0, execution_memory)),
+			      DT_REG_SIZE(DT_INST_PHANDLE(0, execution_memory)))) {
+		LOG_ERR("CPU1 reset pointer 0x%08x invalid.", cpu1_pc);
+		return -EINVAL;
+	} else {
+		LOG_DBG("CPU1 reset pointer: 0x%08x", cpu1_pc);
+	}
+	return 0;
+}
+#endif /* CONFIG_IPM_LOG_LEVEL >= LOG_LEVEL_DBG */
+#endif /* DT_INST_NODE_HAS_PROP(0, execution_memory) */
+
+static int rpi_pico_mailbox_init(const struct device *dev)
+{
+	/* Boot CPU1 if provided a memory region */
+#if DT_INST_NODE_HAS_PROP(0, execution_memory)
+#if DT_INST_NODE_HAS_PROP(0, image_source)
+	BUILD_ASSERT((DT_REG_SIZE(DT_INST_PHANDLE(0, execution_memory)) >=
+		      DT_REG_SIZE(DT_INST_PHANDLE(0, image_source))),
+		     "Image size must not exceed execution memory size");
+
+	BUILD_ASSERT(((DT_REG_ADDR(DT_INST_PHANDLE(0, execution_memory)) >=
+		       DT_REG_ADDR(DT_INST_PHANDLE(0, image_source)) +
+			       DT_REG_SIZE(DT_INST_PHANDLE(0, image_source))) ||
+		      (DT_REG_ADDR(DT_INST_PHANDLE(0, image_source)) >=
+		       DT_REG_ADDR(DT_INST_PHANDLE(0, execution_memory)) +
+			       DT_REG_SIZE(DT_INST_PHANDLE(0, execution_memory)))),
+		     "Image source memory must not overlap with execution memory");
+
+	void *src_mem = (void *)DT_REG_ADDR(DT_INST_PHANDLE(0, image_source));
+	void *exec_mem = (void *)DT_REG_ADDR(DT_INST_PHANDLE(0, execution_memory));
+
+	LOG_DBG("Copying image from %p to %p", src_mem, exec_mem);
+
+	memcpy(exec_mem, src_mem,
+	       MIN(DT_REG_SIZE(DT_INST_PHANDLE(0, execution_memory)),
+		   DT_REG_SIZE(DT_INST_PHANDLE(0, image_source))));
+#endif /* DT_INST_NODE_HAS_PROP(0, image_source) */
+
+	uint32_t cpu1_image_base = DT_REG_ADDR(DT_INST_PHANDLE(0, execution_memory));
+
+	uint32_t *cpu1_vector_table = (void *)cpu1_image_base;
+	uint32_t cpu1_sp = cpu1_vector_table[0];
+	uint32_t cpu1_pc = cpu1_vector_table[1];
+
+#if CONFIG_IPM_LOG_LEVEL >= LOG_LEVEL_DBG
+	if (rpi_pico_validate_image(cpu1_sp, cpu1_pc) != 0) {
+		return -EINVAL;
+	}
+	LOG_DBG("Launching CPU1 with vector table at 0x%p", (void *)cpu1_vector_table);
+#endif /* CONFIG_IPM_LOG_LEVEL >= LOG_LEVEL_DBG */
+
+	if (rpi_pico_reset_cpu1() != 0) {
+		LOG_ERR("CPU1 reset failed.");
+		return -EIO;
+	}
+
+	rpi_pico_boot_cpu1((uint32_t)cpu1_vector_table, cpu1_sp, cpu1_pc);
+#endif /* DT_INST_NODE_HAS_PROP(0, execution_memory) */
+
+	/* Initialize mbox for communication. */
+	IRQ_CONNECT(DT_INST_IRQ_BY_NAME(0, sio_irq_fifo, irq),
+		    DT_INST_IRQ_BY_NAME(0, sio_irq_fifo, priority), rpi_pico_mailbox_isr,
+		    DEVICE_DT_INST_GET(0), 0);
```

Comment:
This is actually the only thing driver does without other changes for second core support, so it's pretty much neutral and mergeable on its own. Anyway, can be disabled by moving the DT addition to the final PR